apps/expert/lib/expert/active_projects.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Expert.ActiveProjects do
+  @moduledoc """
+  A cache to keep track of active projects.
+
+  Since GenLSP events happen asynchronously, we use an ets table to keep track of
+  them and avoid race conditions when we try to update the list of active projects.
+  """
+
+  use GenServer
+
+  def child_spec(_) do
+    %{
+      id: __MODULE__,
+      start: {__MODULE__, :start_link, []}
+    }
+  end
+
+  def start_link do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def init(_) do
+    __MODULE__ = :ets.new(__MODULE__, [:set, :named_table, :public, read_concurrency: true])
+    {:ok, nil}
+  end
```

Comment:
I think wrapping initialization of the ets table in a `Task` should be sufficient since it's owned by no process instead of using a `GenServer` whereby the process would be left alive doing nothing. Or the `init/1` method should return `:ignore` so that the Supervisor does not keep it running, but I'm keen on a `Task` added to the supervision tree instead.

Reply:
Please ignore my comment. 😅